### PR TITLE
[crypto] remove snarkydef (part 3/3 to remove snarky AST)

### PR DESCRIPTION
**Pull request by mimoo**
the other side of https://github.com/o1-labs/snarky/pull/761

builds on top of https://github.com/MinaProtocol/mina/pull/12347

### CODE REVIEW

**Comment by mimoo** [comment]
!ci-build-me

File: src/lib/consensus/proof_of_stake.ml [hunk 1]
```diff
@@ -3419,10 +3438,15 @@ module Make_str (A : Wire_types.Concrete) = struct
...
-        let%snarkydef.Tick next_state_checked ~constraint_constants
+        let next_state_checked ~constraint_constants
             ~(prev_state : Protocol_state.var)
             ~(prev_state_hash : Mina_base.State_hash.var) transition
             supply_increase =
+          let label =
+            Stdlib.(
+              "next_state_checked: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+          in
+          let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/consensus/proof_of_stake.ml:3451]: Ditto?

File: src/lib/mina_base/fee_excess.ml [hunk 2]
```diff
@@ -246,10 +246,15 @@ let eliminate_fee_excess (fee_token_l, fee_excess_l) (fee_token_m, fee_excess_m)
...
-let%snarkydef_ eliminate_fee_excess_checked (fee_token_l, fee_excess_l)
+let eliminate_fee_excess_checked (fee_token_l, fee_excess_l)
     (fee_token_m, fee_excess_m) (fee_token_r, fee_excess_r) =
   let open Tick in
   let open Checked.Let_syntax in
+  let label =
+    Stdlib.(
+      "eliminate_fee_excess_checked: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+  in
+  let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/mina_base/fee_excess.ml:257]: Ditto?

File: src/lib/mina_base/fee_excess.ml [hunk 3]
```diff
@@ -426,6 +431,10 @@ let%snarkydef_ combine_checked
     ; fee_excess_r = fee_excess2_r
     } =
   let open Checked.Let_syntax in
+  let label =
+    Stdlib.("combine_checked: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+  in
+  let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/mina_base/fee_excess.ml:437]: Same?

File: src/lib/mina_base/ledger_hash.ml [hunk 4]
```diff
@@ -105,10 +108,14 @@ let%snarkydef_ modify_account ~depth t aid
    - returns a root [t'] of a tree of depth [depth] which is [t] but with the
      account [f account] at path [addr].
 *)
-let%snarkydef_ modify_account_send ~depth t aid ~is_writeable ~f =
+let modify_account_send ~depth t aid ~is_writeable ~f =
+  let label =
+    Stdlib.("modify_account_send: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+  in
+  let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/mina_base/ledger_hash.ml:115]: Here too

File: src/lib/mina_base/ledger_hash.ml [hunk 5]
```diff
@@ -81,8 +81,11 @@ let get ~depth t addr =
    - returns a root [t'] of a tree of depth [depth] which is [t] but with the
      account [f account] at path [addr].
 *)
-let%snarkydef_ modify_account ~depth t aid
-    ~(filter : Account.var -> 'a Checked.t) ~f =
+let modify_account ~depth t aid ~(filter : Account.var -> 'a Checked.t) ~f =
+  let label =
+    Stdlib.("modify_account: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+  in
+  let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/mina_base/ledger_hash.ml:88]: Here too

File: src/lib/mina_base/ledger_hash.ml [hunk 6]
```diff
@@ -136,10 +143,14 @@ let%snarkydef_ modify_account_send ~depth t aid ~is_writeable ~f =
    - returns a root [t'] of a tree of depth [depth] which is [t] but with the
      account [f account] at path [addr].
 *)
-let%snarkydef_ modify_account_recv ~depth t aid ~f =
+let modify_account_recv ~depth t aid ~f =
+  let label =
+    Stdlib.("modify_account_recv: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+  in
+  let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/mina_base/ledger_hash.ml:150]: Here too

File: src/lib/mina_base/pending_coinbase.ml [hunk 7]
```diff
@@ -835,10 +835,14 @@ module Make_str (A : Wire_types.Concrete) = struct
...
-      let%snarkydef_ add_coinbase
+      let add_coinbase
           ~(constraint_constants : Genesis_constants.Constraint_constants.t) t
           ({ action; coinbase_amount = amount } : Update.var) ~coinbase_receiver
           ~supercharge_coinbase state_body_hash =
+        let label =
+          Stdlib.("add_coinbase: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+        in
+        let%bind () = with_label label (fun _ -> return ()) in
         let depth = constraint_constants.pending_coinbase_depth in
```
Comment (rbonichon) [inline src/lib/mina_base/pending_coinbase.ml:846]: Here too

File: src/lib/mina_base/pending_coinbase.ml [hunk 8]
```diff
@@ -959,9 +963,13 @@ module Make_str (A : Wire_types.Concrete) = struct
...
         Hash.var_of_hash_packed root
 
-      let%snarkydef_ pop_coinbases
+      let pop_coinbases
           ~(constraint_constants : Genesis_constants.Constraint_constants.t) t
           ~proof_emitted =
+        let label =
+          Stdlib.("pop_coinbases: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+        in
+        let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/mina_base/pending_coinbase.ml:972]: Here as well

File: src/lib/snarky_curves/snarky_curves.ml [hunk 9]
```diff
@@ -274,8 +276,10 @@ module Make_weierstrass_checked
       (module M : S)
   end
 
-  let%snarkydef_ double (ax, ay) =
+  let double (ax, ay) =
     let open F in
+    let label = Stdlib.("double: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__) in
+    let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/snarky_curves/snarky_curves.ml:282]: Same

File: src/lib/snarky_curves/snarky_curves.ml [hunk 10]
```diff
@@ -184,8 +184,10 @@ module Make_weierstrass_checked
 
   open Let_syntax
 
-  let%snarkydef_ add' ~div (ax, ay) (bx, by) =
+  let add' ~div (ax, ay) (bx, by) =
     let open F in
+    let label = Stdlib.("add': " ^ __FILE__ ^ ":" ^ string_of_int __LINE__) in
+    let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/snarky_curves/snarky_curves.ml:190]: Same

File: src/lib/snarky_curves/snarky_curves.ml [hunk 11]
```diff
@@ -323,12 +327,13 @@ module Make_weierstrass_checked
...
 
-  let%snarkydef_ scale (type shifted)
-      (module Shifted : Shifted.S with type t = shifted) t
+  let scale (type shifted) (module Shifted : Shifted.S with type t = shifted) t
       (c : Boolean.var Bitstring_lib.Bitstring.Lsb_first.t) ~(init : shifted) :
       shifted Checked.t =
     let c = Bitstring_lib.Bitstring.Lsb_first.to_list c in
     let open Let_syntax in
+    let label = Stdlib.("scale: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__) in
+    let%bind () = with_label label (fun _ -> return ()) in
```
Comment (rbonichon) [inline src/lib/snarky_curves/snarky_curves.ml:336]: Again

File: src/lib/blockchain_snark/blockchain_snark_state.ml [hunk 12]
```diff
@@ -101,10 +101,11 @@ let non_pc_registers_equal_var t1 t2 =
         transition consensus data is valid
         new consensus state is a function of the old consensus state
 *)
-let%snarkydef_ step ~(logger : Logger.t)
-    ~(proof_level : Genesis_constants.Proof_level.t)
+let step ~(logger : Logger.t) ~(proof_level : Genesis_constants.Proof_level.t)
     ~(constraint_constants : Genesis_constants.Constraint_constants.t) new_state
     : _ Tick.Checked.t =
+  let label = Stdlib.("step: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__) in
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/blockchain_snark/blockchain_snark_state.ml:107]: I don't think you quite achieve the same effect here as with `let%snarkydef_` at this point

File: src/lib/transaction_snark/test/multisig_account/multisig_account.ml [hunk 13]
```diff
@@ -33,20 +33,30 @@ let%test_module "multisig_account" =
...
         | [] ->
             Checked.return ()
 
-      let%snarkydef_ distinct_public_keys x = distinct_public_keys x
+      let distinct_public_keys x =
+        let label =
+          Stdlib.(
+            "distinct_public_keys: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+        in
+        let%bind () = with_label label (fun _ -> Checked.return ()) in
```
Comment (rbonichon) [inline src/lib/transaction_snark/test/multisig_account/multisig_account.ml:46]: Same thing. 

However, now I'm wondering why you even need the `let%bind`. You could basically do something like `Format.printf "distinct_public_keys: %s : %d@." __FILE__  __LINE__` at this point.

File: src/lib/blockchain_snark/blockchain_snark_state.ml [hunk 14]
```diff
@@ -101,10 +101,11 @@ let non_pc_registers_equal_var t1 t2 =
...
+  let%bind () = with_label label (fun _ -> Checked.return ()) in
```
Comment (mrmr1993) [inline src/lib/blockchain_snark/blockchain_snark_state.ml:108]: The whole computation needs to be wrapped in the `with_label`. As written, this has no effect.

File: src/lib/consensus/proof_of_stake.ml [hunk 15]
```diff
@@ -738,24 +743,24 @@ module Make_str (A : Wire_types.Concrete) = struct
                 ledger staker_addr )
         in
         let%bind () =
-          [%with_label_ "Account is for the default token"] (fun () ->
+          with_label "Account is for the default token" (fun () ->
```
Comment (mrmr1993) [inline src/lib/consensus/proof_of_stake.ml:746]: Revert. We agreed to get rid of `snarkydef`, not the whole of snarky's ppx.

Comment (mimoo) [reply src/lib/consensus/proof_of_stake.ml:746]: oh, I thought that's what we meant

File: src/lib/consensus/proof_of_stake.ml [hunk 16]
```diff
@@ -2191,6 +2206,10 @@ module Make_str (A : Wire_types.Concrete) = struct
           ~constraint_constants
           ~(protocol_constants : Mina_base.Protocol_constants_checked.var) =
         let open Snark_params.Tick in
+        let label =
+          Stdlib.("update_var: " ^ __FILE__ ^ ":" ^ string_of_int __LINE__)
+        in
+        let%bind () = with_label label (fun _ -> Checked.return ()) in
```
[the remark this answers is not in the document]
Comment (o1-david) [reply src/lib/consensus/proof_of_stake.ml:2212]: in retrospect that idea makes a lot of sense : D

**Comment by mimoo** [comment]
ended up with a simpler approach in https://github.com/MinaProtocol/mina/pull/12365

**Pull request closed without merging**